### PR TITLE
Fix for blank django form fields when VectorField is not required

### PR DESCRIPTION
**Pull request by Cdingram**
If you have a non-required VectorField in django (null=True/blank=True), the saving or creation of a model with a django form would error when trying to save it as blank. This was due to the inherited functionality from forms.CharField. If you look at how [JSONFields](https://github.com/django/django/blob/d59066b90c2ef91a956c6682ed3d25bbb31bf216/django/forms/fields.py#L1371) (which also inherit CharField) they do the same thing as this fix for empty_values. You can try my test case before/after the change in forms.py.

### CODE REVIEW

File: pgvector/django/forms.py [hunk 1]
```diff
@@ -10,3 +10,8 @@ def has_changed(self, initial, data):
         if isinstance(initial, np.ndarray):
             initial = initial.tolist()
         return super().has_changed(initial, data)
+
+    def to_python(self, value):
+        if value in self.empty_values:
```
Comment (ankane) [inline pgvector/django/forms.py:15]: I think `isinstance(value, str) and value == ''` would be a better check so `[]` is still treated as a vector. Also, please keep the trailing newline.

File: tests/test_django.py [hunk 2]
```diff
@@ -59,6 +65,13 @@ class Migration(migrations.Migration):
                 ('embedding', pgvector.django.VectorField(dimensions=3, null=True)),
             ],
         ),
+        migrations.CreateModel(
+            name='ItemBlankAndNull',
+            fields=[
+                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('embedding', pgvector.django.VectorField(dimensions=3, null=True, blank=True)),
```
Comment (ankane) [inline tests/test_django.py:72]: Let's add `blank=True` to the existing field so a new model isn't needed.

**Review by ankane — COMMENTED** [review]
Thanks @Cdingram. Added a few comments inline.

**Comment by Cdingram** [comment]
Thanks, let me know if this looks better.

**Comment by ankane** [comment]
Thanks @Cdingram, looks good without the last commit.

**Comment by Cdingram** [comment]
Yeah sorry github made me think there was no space. Reverted it.

**Comment by ankane** [comment]
Looks great, thanks!

**Pull request merged**